snippets/rust/rust.json

```diff
@@ -317,7 +357,10 @@
     "match": {
         "prefix": "match",
         "body": [
-            "match ${1:expr} {}"
+            "match ${1:expr} {",
```

Comment:
If anything, we should add a `${0}` inside the match curly braces and not refill the match arms. This will provide greater flexibility for end users.

Reply:
Agree, but why `${0}` instead of `${2}`?

Reply:
> Agree, but why `${0}` instead of `${2}`?

The final location of the cursor should be indexed zero. I don't know why but everyone else does it.

Reply:
> I don't know why but everyone else does it.

Yeah, same here.

Functionally `${2}` would accomplish our goal here, but `${0}` is guaranteed to be the final location of the cursor.

Reply:
Sorry if I wasn't clear, I understand that the final location is marked with `${0}`, but my question is why would we want to make the final location be inside the match expression instead of outside of it? As I see it, I'd expect the cursor locations to be:

1. The expression to match
2. Inside the curly braces (to write the "arms" of the match)
3. Outside the match so I can continue writing the rest of my code

Also, as I understand it, isn't `${0}` implicitly place at the end of the snippet and you would only include it if you want the final location to be anywhere else?

Reply:
There is a r-a code action for this, we should delete this snippet.